plugin-flex-ts-template-v2/src/feature-library/custom-transfer-directory/custom-components/QueueDirectoryTab.tsx

```diff
@@ -0,0 +1,286 @@
+  ITask,
+  Notifications,
+} from '@twilio/flex-ui';
+import { useEffect, useState, useRef } from 'react';
+
+import { getAllSyncMapItems } from '../../../utils/sdk-clients/sync/SyncClient';
+import { SearchBox } from './CommonDirectoryComponents';
+import { QueueItem } from './QueueItem';
+import {
+  showOnlyQueuesWithAvailableWorkers,
+  shouldFetchInsightsData,
+  enforceQueueFilterFromWorker,
+  getGlobalFilter,
+  shouldEnforceGlobalFilter,
+} from '../config';
+import { CustomTransferDirectoryNotification } from '../flex-hooks/notifications/CustomTransferDirectory';
+import { CustomWorkerAttributes } from '../../../types/task-router/Worker';
+
+export interface IRealTimeQueueData {
+  total_tasks: number | null;
+  longest_task_waiting_age: number | null;
+  tasks_by_status: {
+    wrapping: number;
+    reserved: number;
+    pending: number;
+    assigned: number;
+  } | null;
+  total_eligible_workers: number | null;
+  total_available_workers: number | null;
+}
+
+export interface TransferQueue extends IQueue, IRealTimeQueueData {}
+
+export interface TransferClickPayload {
+  mode: 'WARM' | 'COLD';
+}
+
+export interface OwnProps {
+  task: ITask;
+}
+
+export interface MapItem {
+  data: object | IRealTimeQueueData;
+  key: string;
+}
+
+const mapRealTimeDataToTransferQueueItem = (
+  transferQueue: TransferQueue,
+  queueData?: IRealTimeQueueData,
+): TransferQueue => {
+  transferQueue.total_eligible_workers = queueData ? queueData.total_eligible_workers : null;
+  transferQueue.total_available_workers = queueData ? queueData.total_available_workers : null;
+  transferQueue.total_tasks = queueData ? queueData.total_tasks : null;
+  transferQueue.longest_task_waiting_age = queueData ? queueData.longest_task_waiting_age : null;
+  transferQueue.tasks_by_status = queueData ? queueData.tasks_by_status : null;
+
+  return transferQueue;
+};
+
+const QueueDirectoryTab = (props: OwnProps) => {
+  const [fetchedQueues, setFetchedQueues] = useState([] as Array<IQueue>);
+  const [insightsQueues, setInsightsQueues] = useState([] as Array<MapItem>);
+  const [filteredQueues, setFilteredQueues] = useState([] as Array<TransferQueue>);
+  const [queueFilterTimer, setQueueFiltertimer] = useState(null as any);
+
+  const transferQueues = useRef([] as Array<TransferQueue>);
+  const searchInputRef = useRef<HTMLInputElement>(null);
+
+  const { workspaceClient, insightsClient, workerClient } = Manager.getInstance();
+
+  // takes the input in the search box and applies it to the queue result
+  // this will trigger the useEffect for a queueFilter update
+  const onQueueSearchInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (!fetchedQueues || queueFilterTimer) {
+      return;
+    }
+
+    setQueueFiltertimer(
+      setTimeout(() => {
+        // eslint-disable-next-line no-eq-null, eqeqeq
+        if (event.target != null) {
+          filterQueues();
+          setQueueFiltertimer(null);
+        }
+      }, 300),
```

Comment:
Is this basically just a debounce? It might be easier to understand to use the lodash `debounce` function which we already have available for use in the template.